File: storage/src/main/java/org/epdb/storage/PageConstants.java
package org.epdb.storage;

public class PageConstants {
    public static final int PAGE_SIZE_IN_BYTES = 4096;
    public static final int HEADER_SIZE_IN_BYTES = 8;
    public static final int HEADER_FREE_SPACE_OFFSET_ADDR = 0;
    public static final int HEADER_NUM_ROWS_ADDR = 4;
    public static final int SLOT_SIZE_IN_BYTES = 8;
    public static final int SLOT_RECORD_OFFSET_SIZE_IN_BYTES = 4;
    public static final int ROW_SIZE_IN_BYTES = 28;
    public static final int HEADER_NEXT_PAGE_ID_ADDR = 8;
    public static final int NO_NEXT_PAGE = -1;
}

File: storage/src/main/kotlin/org/epdb/storage/util/PageConstants.kt
package org.epdb.storage.util

object PageConstants {
    const val PAGE_SIZE_IN_BYTES: Int = 4096
    const val HEADER_SIZE_IN_BYTES: Int = 12
    const val HEADER_FREE_SPACE_OFFSET_ADDR: Int = 0
    const val HEADER_NUM_ROWS_ADDR: Int = 4
    const val HEADER_NEXT_PAGE_ID_ADDR: Int = 8
    const val SLOT_SIZE_IN_BYTES: Int = 8
    const val SLOT_RECORD_OFFSET_SIZE_IN_BYTES: Int = 4
    const val ROW_SIZE_IN_BYTES: Int = 28
    const val NO_NEXT_PAGE: Int = -1
}